sktime/regression/deep_learning/base.py

```diff
@@ -187,15 +186,11 @@ def save(self, path=None):
         from zipfile import ZipFile
 
         if path is None:
-            _check_soft_dependencies("h5py")
-            import h5py
-
             in_memory_model = None
             if self.model_ is not None:
-                self.model_.save("disk_less.h5")
-                with h5py.File("disk_less.h5", "r") as h5file:
-                    in_memory_model = h5file.id.get_file_image()
-
+                model_json = self.model_.to_json()
```

Comment:
this is a different change from vendoring, no?

Reply:
Oh true, I directly tried to fix the serialisation problems and committed it.

Reply:
should we split the vendoring and serialisation into two PR?

The current imports are only in `networks` afaik, so it can be split easily.